locations/spiders/tjmaxx.py

```diff
@@ -0,0 +1,131 @@
+                'ref': link_id,
+                'opening_hours': self.process_hours(hours[0])
+            }
+
+        lon_lat = [
+            float(longitude[0]),
+            float(latitude[0]),
+        ]
+
+        yield GeojsonPointItem(
+            properties = properties,
+            lon_lat = lon_lat,
+        )
+
+
+    def process_hours(self, hours):
+        """ This should capture these time formats as on tjmaxx
+                # Mon-Wed: 9:30a-9:30p,
+                # Thanksgiving Day: CLOSED,  <- this gets appended without modification
+                # Fri-Sat: 7a-10p,
+                # Sun: 9a-10p
+        :param hours:
+        :return:  string - in this format Mo-Th 11:00-12:00; Fr-Sa 11:00-01:00;
+        """
+        working_hour = []
+
+        for t in hours.split(","):
+            match = re.search(r"^((\w{2})\w(?:-(\w{2})\w)?\:\s(\d+)(\:\d+)?(\w)-(\d+)(\:\d+)?(\w))$",
+                              t.strip(),
+                              re.IGNORECASE)
+            if match:
+                m = list(match.groups())
+                if len(m) == 9:
+                    # replace second entry for day of week for formats like Sun: 9a-12p
+                    if m[2] is None:
+                        m[2] = ""
+                    else:
+                        m[2] = "-" + m[2]
+
+                    # concatenate to our format
+                    if m[4] is None:
+                        if m[7] is None:
+                            # time is like Mon-Wed: 9a-9p
+                            working_hour.append(m[1] + m[2] + " " + str(int(m[3]) + self.am_pm(m[3], m[5])) + ":00-" +
+                                                str(int(m[6]) + self.am_pm(m[6], m[8])) + ":00")
+                        else:
+                            # time is like Mon-Wed: 9a-9:30p
+                            working_hour.append(m[1] + m[2] + " " + str(int(m[3]) + self.am_pm(m[3], m[5])) + ":00-" +
+                                                str(int(m[6]) + self.am_pm(m[6], m[8])) + m[7])
+                    else:
+                        if m[7] is None:
+                            # time is like Mon-Wed: 9:30a-9p
+                            working_hour.append(m[1] + m[2] + " " + str(int(m[3]) + self.am_pm(m[3], m[5])) + m[4] +
+                                                "-" + str(int(m[6]) + self.am_pm(m[6], m[8])) + ":00")
+                        else:
+                            # time is like Mon-Wed: 9:30a-9:30p
+                            working_hour.append(m[1] + m[2] + " " + str(int(m[3]) + self.am_pm(m[3], m[5])) + m[4] +
+                                                "-" + str(int(m[6]) + self.am_pm(m[6], m[8])) + m[7])
+            # else:
+            #     working_hour.append(t)
+
+        if len(working_hour):
+            return "; ".join(working_hour)
+        else:
+            # if something fails, return original date
+            return hours
+
+    def am_pm(self, m3, x):
```

Comment:
Can you use more descriptive argument names here to describe what these things represent?